### PR TITLE
Fix push tests by adding user.name and user.email to the commit

### PR DESCRIPTION
**Pull request by JimSuplizio**
The push tests, when running on in automation, were failing because the commit message didn't have the user.name or user.email set when running on the lab machines. 
1. Change the token environment variable to GIT_TOKEN
2. Add GIT_COMMIT_OWNER and GIT_COMMIT_EMAIL as environment variables to the server in tests.yml
3. If neither of the environment variables in step 2 are set, check git for these values. If those aren't set, I'm probably going to have to prompt the user to enter them. That is still to be decided and part of this PR.

Right now, this PR is to unblock the failures that are happening when running **tools - test-proxy - tests** pipeline.

### CODE REVIEW

**Comment by JimSuplizio** [comment]
@scbedd I ran the [tools - test-proxy - tests](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1848527&view=logs&j=feaa6207-9943-5fce-22b1-ded3f19fedb9&t=1858cc7c-ecda-5ece-3e3c-11fecc646260) pipeline against the refs/merge and look, everything now passes.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/GitStore.cs [hunk 1]
```diff
@@ -32,7 +32,11 @@ public class GitStore : IAssetsStore
         public GitProcessHandler GitHandler = new GitProcessHandler();
         public string DefaultBranch = "main";
         public string FileName = "assets.json";
-        public static readonly string EnvironmentVariableName = "PROXY_GIT_TOKEN";
+        public static readonly string GIT_TOKEN_ENV_VAR = "GIT_TOKEN";
+        // Note: These are slightly different from the GIT_COMMITTER_NAME and GIT_COMMITTER_EMAIL
```
Comment (scbedd) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/GitStore.cs:36]: We should discuss with Wes and Ben as to their opinions on not overloading the defaults. I'm good with merging this though.

Comment (scbedd) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/GitStore.cs:36]: Variable names are easy to change when this isn't being actively used :)

**Review by scbedd — APPROVED** [review]
👍

**Pull request merged**